### PR TITLE
Correctly display current values for indexed options in help.

### PR DESCRIPTION
**Pull request by dwsteele**
The current value for an indexed option was always for the first index, e.g. pg1-path. This is likely legacy from before indexing was added (and faithfully copied over from Perl, apparently).

Fix this by enumerating the current values in the option help and displaying <multi> in the option list when more than one value exists.

### CODE REVIEW

**Comment by dwsteele** [comment]
@pgstef @pgguru @cbandy This addresses the issues with current values brought up in #2395. However, instead of just removing them, I fixed them.

File: src/command/help/help.c [hunk 1]
```diff
@@ -224,6 +224,42 @@ helpRenderValue(const ConfigOption optionId, const unsigned int optionIdx)
...
+            for (unsigned int optionIdx = 1; optionIdx < cfgOptionGroupIdxTotal(cfgOptionGroupId(optionId)); optionIdx++)
+            {
+                const String *const value = helpRenderValueIdx(optionId, optionIdx);
+
+                if (!strEq(result, value))
+                {
+                    MEM_CONTEXT_PRIOR_BEGIN()
+                    {
+                        strFree(result);
+                        result = strNewZ("<multi>");
```
Comment (pgguru) [inline src/command/help/help.c:249]: So this is the ultimate output if the values are different?  I guess I'd assumed it'd show all of the values in this case, but if we're able to query individually it's probably fine.

Comment (pgguru) [reply src/command/help/help.c:249]: (I see later that this is a sentinel value, so disregard.)

File: test/src/module/command/helpTest.c [hunk 2]
```diff
@@ -584,6 +585,61 @@ testRun(void)
...
+        hrnCfgArgKeyRawZ(argList, cfgOptRepoHost, 1, "backup.example.net");
+        hrnCfgArgKeyRawZ(argList, cfgOptRepoHost, 3, "dr.test.org");
+        strLstAddZ(argList, "help");
+        strLstAddZ(argList, "restore");
+        strLstAddZ(argList, "repo-host");
+        TEST_RESULT_VOID(testCfgLoad(argList), "help for restore command, repo-host option");
+        TEST_RESULT_STR_Z(helpRender(helpData), optionHelp, "check text");
+
+        // -------------------------------------------------------------------------------------------------------------------------
+        TEST_TITLE("multiple current values (with one unset)");
```
Comment (pgguru) [inline test/src/module/command/helpTest.c:618]: Not sure I understand the "unset" option here—just showing repo3's option?  Is the default for other repos supposed to be shown as well?

Comment (dwsteele) [reply test/src/module/command/helpTest.c:618]: Just to make sure gaps are rendered correctly, e.g. `repo1-path`, `repo3-path`.

Comment (dwsteele) [reply test/src/module/command/helpTest.c:618]: In other words, repo2 is skipped.

Comment (pgguru) [reply test/src/module/command/helpTest.c:618]: Is it possible to set a global default per setting and override it on a repo-by-repo basis?  Is there a test exercising that if so? (Didn't seem to be in the new ones, which I'd expect if we're outputting for all of the known repos.)

Comment (dwsteele) [reply test/src/module/command/helpTest.c:618]: > s it possible to set a global default per setting and override it on a repo-by-repo basis?

**Pull request merged**